src/swapRouter.ts

```diff
@@ -9,25 +9,82 @@ import { NFTTrade, SupportedProtocolsData } from './entities/NFTTrade'
 import { UniswapTrade, SwapOptions } from './entities/protocols/uniswap'
 import { CommandType, RoutePlanner } from './utils/routerCommands'
 import { encodePermit } from './utils/permit2'
-import { ETH_ADDRESS } from './utils/constants'
+import { ROUTER_AS_RECIPIENT, MSG_SENDER, ETH_ADDRESS } from './utils/constants'
 
 export type SwapRouterConfig = {
   sender?: string // address
   deadline?: BigNumberish
 }
 
+type SupportedNFTTrade = NFTTrade<SupportedProtocolsData>
+
 export abstract class SwapRouter {
   public static INTERFACE: Interface = new Interface(abi)
 
-  /**
+  public static swapCallParameters(
+    trades: (SupportedNFTTrade | UniswapTrade)[] | SupportedNFTTrade | UniswapTrade,
+    config: SwapRouterConfig = {}
+  ): MethodParameters {
+    if (!Array.isArray(trades)) trades = [trades]
+
+    const nftTrades = trades.filter((trade, index, []) => trade instanceof NFTTrade) as SupportedNFTTrade[]
+    const allowRevert = nftTrades.length == 1 && nftTrades[0].orders.length == 1 ? false : true
+    const planner = new RoutePlanner()
+
+    // track value flow to require the right amount of native value
+    let currentNativeValueInRouter = BigNumber.from(0)
+    let transactionValue = BigNumber.from(0)
+
+    for (const trade of trades) {
+      if (trade instanceof NFTTrade) {
```

Comment:
Do you know what was causing this? Was it an object clash with the Token object or something?

Reply:
Oh, maybe! I don't think we were entirely sure what caused it, since the shared-dep versions were matched up. @zzmp @tinaszheng can you double-check here? I personally would just lean towards generally avoiding instanceof in sdks after our prev issues

Reply:
Ooh yeah instanceof checks don't work across packages unless you import this as a peer dependency but I think the better workaround is to check for type properties instead of relying on instanceof checks

Reply:
+1 good rule of thumb is just to never use `instanceof` in TS, and always use discriminant unions